src/cli/uuid.test.ts

```diff
@@ -0,0 +1,146 @@
+      items: [
+        {
+          metadata: {
+            name: "test-deploy",
+            labels: { "pepr.dev/uuid": "1234" },
+            annotations: { "pepr.dev/description": "Test annotation" },
+          },
+        },
+      ],
+    };
+
+    const result = buildUUIDTable(deployments);
+    expect(result).toEqual({ "1234": "Test annotation" });
+  });
+
+  it("should include UUID with empty description if none is present", () => {
+    const deployments: KubernetesListObject<kind.Deployment> = {
+      apiVersion: "v1",
+      kind: "List",
+      items: [
+        {
+          metadata: {
+            name: "no-desc",
+            labels: { "pepr.dev/uuid": "5678" },
+          },
+        },
+      ],
+    };
+
+    const result = buildUUIDTable(deployments);
+    expect(result).toEqual({ "5678": "" });
+  });
+
+  it("should skip deployments without UUID", () => {
+    const deployments: KubernetesListObject<kind.Deployment> = {
+      apiVersion: "v1",
+      kind: "List",
+      items: [
+        {
+          metadata: {
+            name: "no-uuid",
+            annotations: { "pepr.dev/description": "Should be ignored" },
+          },
+        },
+      ],
+    };
+
+    const result = buildUUIDTable(deployments);
+    expect(result).toEqual({});
+  });
+});
+describe("getPeprDeploymentsByUUID", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("should fetch deployments without uuid", async () => {
+    const result = await getPeprDeploymentsByUUID();
+
+    expect(result.items[0].metadata?.labels?.["pepr.dev/uuid"]).toBe("1234");
+    expect(result.items[0].metadata?.annotations?.["pepr.dev/description"]).toBe("Test annotation");
+  });
+
+  it("should fetch deployments with a specific uuid", async () => {
+    const result = await getPeprDeploymentsByUUID("abcd-uuid");
+
+    expect(result.items[0].metadata?.labels?.["pepr.dev/uuid"]).toBe("1234");
+  });
+});
+
+describe("uuid CLI command", () => {
+  let program: Command;
+  let logSpy: ReturnType<typeof vi.spyOn>;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    program = new Command();
+    uuid(program);
+    logSpy = vi.spyOn(console, "log").mockImplementation(() => {});
+  });
+
+  it("should log UUID and description when deployment has a UUID", async () => {
+    await program.parseAsync(["uuid"], { from: "user" });
+
+    expect(logSpy).toHaveBeenCalledWith("UUID\t\tDescription");
+    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
+    expect(logSpy).toHaveBeenCalledWith("1234\tTest annotation");
+  });
+
+  it("should log UUID and description when deployment has a matching UUID", async () => {
+    await program.parseAsync(["uuid", "asdf"], { from: "user" });
+    expect(logSpy).toHaveBeenCalledWith("UUID\t\tDescription");
+    expect(logSpy).toHaveBeenCalledWith("--------------------------------------------");
+    expect(logSpy).toHaveBeenCalledWith("asdf\tAnother annotation");
```

Comment:
```suggestion
    expect(logSpy).not.toHaveBeenCalledWith("1234\tTest annotation");
    expect(logSpy).toHaveBeenCalledWith("asdf\tAnother annotation");
```

This test case focuses on providing a UUID to filter a collection of deployments. So, given a mock that produces two deployments, we should be able to filter it down to just one (`asdf`) to make sure that providing a UUID filters as expected.

Note that current state of the PR causes the test to fail. So, we might need to revisit the implementation details of `getPeprDeploymentsByUUID()`

Reply:
So you are right, and in an integration test that would work but here it is not going to work based on how the mock is set up

```bash
logSpy calls:
[
  [
    "UUID\t\tDescription"
  ],
  [
    "--------------------------------------------"
  ],
  [
    "1234\tTest annotation"
  ],
  [
    "asdf\tAnother annotation"
  ]
]
```

The deployment is always returning two items

```typescript
  const Get = vi.fn().mockResolvedValue({
    items: [
      {
        metadata: {
          name: "test-deployment",
          labels: { "pepr.dev/uuid": "1234" },
          annotations: { "pepr.dev/description": "Test annotation" },
        },
      },
      {
        metadata: {
          name: "other-deployment",
          labels: { "pepr.dev/uuid": "asdf" },
          annotations: { "pepr.dev/description": "Another annotation" },
        },
      },
    ],
  });
  ```

Reply:
ah i see yeah you had just mentioned what I just said, disregard